src/app/util/ember-compatibility-functions.cpp

```diff
@@ -216,218 +216,221 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
-        case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint32_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint64_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int8_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int16_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int32_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT64S_ATTRIBUTE_TYPE: // Signed 64-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int64_t>(*writer, isNullable));
-            break;
+                // TODO: We should probably clone the writer and convert failures here
+                // into status responses, unless our caller already does that.
+                AttributeValueEncoder valueEncoder(writer, aAccessingFabricIndex);
+                ReturnErrorOnFailure(attrOverride->Read(readPath, valueEncoder));
+
+                if (valueEncoder.TriedEncode())
+                {
+                    status = Protocols::InteractionModel::Status::Success;
+                    ExitNow();
+                }
+            }
         }
-        case ZCL_CHAR_STRING_ATTRIBUTE_TYPE: // Char string
+
+        EmberAfAttributeMetadata * metadata = nullptr;
+        EmberAfAttributeSearchRecord record;
+        record.endpoint           = aPath.mEndpointId;
+        record.clusterId          = aPath.mClusterId;
+        record.clusterMask        = CLUSTER_MASK_SERVER;
+        record.attributeId        = aPath.mAttributeId;
+        record.manufacturerCode   = EMBER_AF_NULL_MANUFACTURER_CODE;
+        EmberAfStatus emberStatus = emAfReadOrWriteAttribute(&record, &metadata, attributeData, sizeof(attributeData),
+                                                             /* write = */ false);
+
+        if (emberStatus == EMBER_ZCL_STATUS_SUCCESS)
         {
-            char * actualData  = reinterpret_cast<char *>(attributeData + 1);
-            uint8_t dataLength = attributeData[0];
-            if (dataLength == 0xFF)
+            EmberAfAttributeType attributeType = metadata->attributeType;
+            bool isNullable                    = metadata->IsNullable();
+            TLV::Tag tag                       = TLV::ContextTag(to_underlying(AttributeDataIB::Tag::kData));
+            switch (BaseType(attributeType))
```

Comment:
Needs merging to the float/double bits that landed recently....

Reply:
I got them now that I merged.